pages/volunteer/signup.tsx

```diff
@@ -183,36 +234,30 @@ export default function VolunteerInfo({
             headers: { "Content-Type": "application/json" },
             body: JSON.stringify(userData),
         };
-        const response = await fetch("api/user", request);
-        const updatedUserData = await response.json();
-        return updatedUserData;
+        const response = await fetch("/api/user", request);
+        if (response.ok) {
+            const updatedUserData = await response.json();
+            return updatedUserData;
+        } else {
+            return null;
+        }
     }
 
-    const clearLocalStorage = () => {
-        setVolunteerFirstName("");
-        setVolunteerLastName("");
-        setPhoneNumber("");
-        setDateOfBirth("");
-        setAddress1("");
-        setCity("");
-        setParticipantProvince(DEFAULT_PROVINCE);
-        setPostalCode("");
-        setSchool("");
-        setSkills("");
-        setHeardFrom("");
-    };
     async function updateUserAndClearForm() {
         const updatedUser = await updateUser();
         if (updatedUser) {
-            clearLocalStorage();
+            setSuccessfulAccountCreation("success");
+            localStorage.clear();
             return updatedUser;
         } else {
+            setSuccessfulAccountCreation("failure");
             return null;
         }
     }
```

Comment:
You might also want to invalidate the me query to let SWR know it's time to refetch the current user. Since our me query SWR call is `const { data, error } = useSWR("/api/user/me", fetcher);`, you would want to do

```typescript
    async function updateUserAndClearForm() {
        const updatedUser = await updateUser();
        if (updatedUser) {
            setSuccessfulAccountCreation("success");
            localStorage.clear();
            // New line to invalidate me query
            mutate("/api/user/me");
            return updatedUser;
        } else {
            setSuccessfulAccountCreation("failure");
            return null;
        }
    }
```


This is especially important once you rebase your branch, (there's me queries everywhere now lol)

Reply:
Sounds good